src/any_agent/callbacks/span_generation/base.py

```diff
@@ -16,12 +16,33 @@ class _SpanGeneration(Callback):
     def __init__(self) -> None:
         self.first_llm_calls: set[int] = set()
 
+    def _serialize_for_attribute(self, data: Any) -> str:
+        """Serialize data for OpenTelemetry attributes, handling various types safely."""
+        if isinstance(data, str):
+            return data
+        try:
+            return json.dumps(data, default=str, ensure_ascii=False)
+        except (TypeError, ValueError):
+            return str(data)
+
+    def _determine_output_type(self, output: Any) -> tuple[str, str]:
+        """Determine output type and serialize the output for attributes."""
+        if isinstance(output, str):
+            try:
+                json.loads(output)
+            except json.JSONDecodeError:
+                return output, "text"
+            else:
+                return output, "json"
+        else:
+            return self._serialize_for_attribute(output), "json"
```

Comment:
💅 I would just return the output_type and not couple this with `_serialize_for_attribute`